src/wazuh_db/wdb_syscollector.c

```diff
@@ -42,19 +42,25 @@ int wdb_netinfo_save(wdb_t * wdb, const char * scan_id, const char * scan_time,
         item_id,
         replace) < 0) {
 
-        return -1;
+        return OS_INVALID;
     }
 
-    return 0;
+    return OS_SUCCESS;
 }
 
 // Insert Network info tuple. Return 0 on success or -1 on error.
 int wdb_netinfo_insert(wdb_t * wdb, const char * scan_id, const char * scan_time, const char * name, const char * adapter, const char * type, const char * state, int mtu, const char * mac, long tx_packets, long rx_packets, long tx_bytes, long rx_bytes, long tx_errors, long rx_errors, long tx_dropped, long rx_dropped, const char * checksum, const char * item_id, const bool replace) {
     sqlite3_stmt *stmt = NULL;
 
+    if (NULL == name) {
+        if(checksum && 0 != strcmp(SYSCOLLECTOR_LEGACY_CHECKSUM_VALUE, checksum)) {
+            wdbi_remove_by_pk(wdb, WDB_SYSCOLLECTOR_NETINFO, item_id);
+        }
```

Comment:
```suggestion
        wdbi_remove_by_pk(wdb, WDB_SYSCOLLECTOR_NETINFO, item_id);
```
the item_id is a sync field, no need to re-identify it, check the call tree. (For legacy code, item_id is null)

Reply:
Idem.

Reply:
ditto